File: Chapter05/Python/monitoredProducts.py
import pandas as pd
import re
import os

os.chdir("<path to ProductionOrders.csv file>")
pattern = re.compile(r"^(561769|561394)(?=(01|22|39)(A|B)\d{1}$)")

def monitoredProducts(sku):

    m = pattern.match(sku)
    if m == True:
        return m.group(0)
    else:
        return "Not Moinitored"

df = pd.read_csv("ProductionOrders.csv")

df["Monitored Products"] = df["SKU"].apply(monitoredProducts)

File: Chapter07/R/monitoredProducts.R
library(tidyverse)
library(stringr)

setwd("C:/Users/ryanw/OneDrive - Diesel Analytics/Professional/Clients/Apress/Book/Part2_ShapingData/Chapter5/Misc")

monitoredProducts <- function(x){
  pattern = "^(561769|561394)(?=(01|22|39)(A|B))"
  mp = str_extract(x, pattern = pattern)
  return_value = ifelse(is.na(mp),"Not Monitored", mp)
  return(return_value)
}

df <- read_csv("ProductionOrders.csv")

df <-
  df %>%
  mutate(`Monitored Products` = monitoredProducts(SKU))
